jsonParser/parser.ts

```diff
@@ -0,0 +1,190 @@
+    VOTANTI = 'VOTANTI'
+}
+
+const enum query {
+    DIPARTIMENTO = 'DIPARTIMENTO',
+    ORGANI = "IN SENO"
+}
+
+class Parser {
+    private info: object;
+    private doc: Target;
+    private fileName: string;
+
+    constructor(fileName: string, dip) {
+
+        this.fileName = fileName;
+
+        this.info = {
+            schede: {},
+            liste: [],
+            eletti: []
+        }
+
+        if (dip === 0)
+            this.doc = new Dipartimento();
+        else
+            this.doc = new Organo();
+
+
+    }
+
+    private write(): void {
+        const data = JSON.stringify(this.info);
+        fs.writeFile('json/' + this.fileName + '.json', data, (errW) => {
+            if (errW) {
+                throw errW;
+            }
+            console.log('JSON data is saved.');
+        });
+    }
+
+    public scrape(): void {
+        fs.readFile('document/' + this.fileName + '.pdf', (errR, buffer) => {
+
+            if (errR)
+                return console.log(errR);
+
+            pdf2table.parse(buffer, (errP, data) => {
+                if (errP)
+                    return console.log(errP);
+
+                this.doc.scrapeLists(this.info, data);
+                let idxList = -1;
+
+                data.forEach(el => {
+                    if (this.doc.checkEletto(el)) {
+                        const eletto = {
+                            nominativo: el[0],
+                            voti: el[1],
+                            lista: this.info['liste'][idxList]
+                        }
+                        this.info['eletti'].push(eletto);
+                    }
+
+
+                    switch (el[0]) {
+                        case schede.BIANCHE:
+                        case schede.NULLE:
+                        case schede.CONTESTATE:
+                            this.info['schede'][el[0]] = el[1];
+                            break;
+                        case elettori.TUTTI:
+                            this.info['elettori'] = el[1];
+                            break;
+                        case elettori.VOTANTI:
+                            this.info['votanti'] = el[1];
+                            break;
+                    }
+                    if (el[0].includes(seggi.SCRUTINATI))
+                        idxList++;
+
+                });
+
+
+
+
+
+                this.write();
+            });
+        });
+    }
+
+
+
+
+}
+
+interface Target {
+    scrapeLists(info: object, data: object): void;
+    scrapeOther(info: object, data: object): void;
+    checkEletto(data: object): boolean;
+}
+
+
+class Dipartimento implements Target {
+    public scrapeLists(info: object, data: object[]): void {
+        for (let i = 0; i < data.length; i++) {
+
+            if (data[i][0].includes(query.DIPARTIMENTO))
+                info['dipartimento'] = data[++i][0];
+
+            if (data[i][0].includes(candidati.LISTE_DIP)) {
+                i = i + 2;
```

Comment:
```suggestion
                i += 2;
```
Questo è questione di stile, ma normalmente "short is better" in questi casi :grin: 
Però è molto soggettiva come cosa, quindi vedi un po' tu